pkg/compactor/shuffle_sharding_planner.go

```diff
@@ -35,12 +53,34 @@ func (p *ShuffleShardingPlanner) Plan(_ context.Context, metasByMinTime []*metad
 	resultMetas := make([]*metadata.Meta, 0, len(metasByMinTime))
 
 	for _, b := range metasByMinTime {
+		blockID := b.ULID.String()
 		if _, excluded := noCompactMarked[b.ULID]; excluded {
 			continue
 		}
 
 		if b.MinTime < rangeStart || b.MaxTime > rangeEnd {
-			return nil, fmt.Errorf("block %s with time range %d:%d is outside the largest expected range %d:%d", b.ULID.String(), b.MinTime, b.MaxTime, rangeStart, rangeEnd)
+			return nil, fmt.Errorf("block %s with time range %d:%d is outside the largest expected range %d:%d", blockID, b.MinTime, b.MaxTime, rangeStart, rangeEnd)
+		}
+
+		lockFileReader, err := p.bkt.Get(p.ctx, path.Join(blockID, BlockLockFile))
```

Comment:
maybe we can extract these block download/deserializtion logic to another .go file so it can be shared between the planner and grouper